### DISCUSSION TITLE
Can localstorage be accessed outside useEffect

### DISCUSSION THREAD

**Discussion by bk973** [vercel/next.js#40003] [Help]
It turns out that local storage can be accessed outside useEffect() when using pure react yet this isn't the case with Next Js. I am facing this issue in my latest project and this is the first time am using Next Js

![Capture_local_storage_error](https://user-images.githubusercontent.com/34472816/187024064-055af3ac-1524-4338-bb04-443fccef04dd.PNG)

**Comment by thanhtutzaw** [comment]
Next. js performs a server render before the client render.
Check full article [here](https://developer.school/snippets/react/localstorage-is-not-defined-nextjs#:~:text=This%20is%20because%20localStorage%20is,window%20object%20has%20been%20defined.)

or Same discussions [here](https://github.com/vercel/next.js/discussions/19911)

**Reply by bk973** [reply]
Thank you so much @thanhtutzaw

**Comment by icyJoseph** [comment]
Hi,

This is a common pitfall encountered when switching from CSR to SSR world. Next.js runs also on the server, where window is not available, and in fact, there's no browser. This happens both at build and runtime.

In general what you want to do with SSR applications, is avoid unguarded DOM/window checks, and make sure your logic is sound for each guarded scenario.
